backend/src/db/database.ts

```diff
@@ -1,17 +1,24 @@
-import { Kysely, PostgresDialect } from "kysely";
-import { pool } from "./dbconfig";
-import { Database } from "./types";
-export async function testDbConnection() {
-    const res = await pool.query('SELECT NOW()');
-    if (res) {
-        console.log("DB connection succeeded.");
-    } else {
-        throw Error("DB connection failed.");
-    }
-}
+import { Kysely, PostgresDialect } from 'kysely';
+import { DB } from 'kysely-codegen';
+import pool from './dbconfig.js';
 
-const dialect = new PostgresDialect({ pool: pool });
+export const testDbConnection = () => {
```

Comment:
I tried making this an async function and then using at the top-level of `index.js` and it says it's not allowed in cjs output format. This led me to add `"type": "module",` to my `package.json` because i read that would make the entry be an esm module coupled with my other `tsconfig.json` settings. but then i get a problem with the import of `Pool` from `pg`. This [SO post](https://stackoverflow.com/questions/55377103/can-i-import-the-node-postgres-module-pg-or-is-it-commonjs-only) describes what i'm experiencing but when i try updating the import like it says, i get an error that `TypeError: pg.Pool is not a constructor`. ☹️